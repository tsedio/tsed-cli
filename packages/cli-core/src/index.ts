import "./utils/patchCommander.js";

import Inquirer from "inquirer";

export * from "./CliCore.js";
export * from "./decorators/index.js";
export * from "./interfaces/index.js";
export * from "./packageManagers/index.js";
export * from "./services/index.js";
export * from "./utils/index.js";
export * from "@tsed/core";
export * from "@tsed/di";
export * from "@tsed/logger";
export * from "@tsed/normalize-path";
export {Inquirer};
