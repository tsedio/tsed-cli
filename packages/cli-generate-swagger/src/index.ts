export * from "./commands/GenerateSwaggerCmd";

export default {};
