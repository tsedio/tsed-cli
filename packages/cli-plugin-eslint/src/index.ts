import "./templates/eslint.template.js";

import {CliPluginEslintModule} from "./CliPluginEslintModule.js";
export * from "./utils/templateDir.js";

export default CliPluginEslintModule;
