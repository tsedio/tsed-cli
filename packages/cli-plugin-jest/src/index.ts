import {CliPluginJestModule} from "./CliPluginJestModule.js";

export * from "./templates/index.js";

export default CliPluginJestModule;
