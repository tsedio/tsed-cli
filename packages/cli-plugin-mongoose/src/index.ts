import {CliPluginMongooseModule} from "./CliPluginMongooseModule.js";

export * from "./hooks/MongooseGenerateHook.js";
export * from "./hooks/MongooseInitHook.js";
export * from "./services/CliMongoose.js";

export default CliPluginMongooseModule;
