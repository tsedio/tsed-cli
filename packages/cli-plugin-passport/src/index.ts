import "./templates/index.js";

import {CliPluginPassportModule} from "./CliPluginPassportModule.js";

export * from "./services/PassportClient.js";

export default CliPluginPassportModule;
