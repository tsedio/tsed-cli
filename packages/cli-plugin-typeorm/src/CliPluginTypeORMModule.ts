import type {AlterPackageJson, RenderDataContext} from "@tsed/cli";
import {inject, ProjectPackageJson} from "@tsed/cli-core";
import {injectable} from "@tsed/di";

import {TypeORMGenerateHook} from "./hooks/TypeORMGenerateHook.js";
import {TypeORMInitHook} from "./hooks/TypeORMInitHook.js";

export class CliPluginTypeORMModule implements AlterPackageJson {
  protected packageJson = inject(ProjectPackageJson);

  $alterPackageJson(packageJson: ProjectPackageJson, data: RenderDataContext) {
    if (data.typeorm) {
      packageJson.addDependencies({
        typeorm: "latest"
      });
    }

    return packageJson;
  }
}

injectable(CliPluginTypeORMModule).imports([TypeORMInitHook, TypeORMGenerateHook]);
