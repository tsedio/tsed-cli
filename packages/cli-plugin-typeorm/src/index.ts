import "./templates/datasource.template.js";

import {CliPluginTypeORMModule} from "./CliPluginTypeORMModule.js";

export * from "./hooks/TypeORMGenerateHook.js";
export * from "./hooks/TypeORMInitHook.js";

export default CliPluginTypeORMModule;
