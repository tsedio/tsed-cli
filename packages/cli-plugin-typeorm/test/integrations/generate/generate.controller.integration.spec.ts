import {GenerateCmd, TEMPLATE_DIR} from "@tsed/cli";
import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";
import "../../../src";


describe("Generate DataSource", () => {
  beforeEach(() =>
    CliPlatformTest.bootstrap({
      templateDir: TEMPLATE_DIR,
      commands: [GenerateCmd]
    })
  );
  afterEach(() => CliPlatformTest.reset());

  it("should generate the template with the right options (simple path)", async () => {
    CliPlatformTest.setPackageJson({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await CliPlatformTest.exec("generate", {
      rootDir: "./project-data",
      type: "typeorm:datasource",
      name: "Test",
      typeormDataSource: "typeorm:mysql"
    });

    expect(FakeCliFs.getKeys()).toEqual([
      "project-name/docker-compose.yml",
      "project-name/src/datasources",
      "project-name/src/datasources/TestDatasource.ts"
    ]);

    const datasource = FakeCliFs.entries.get("project-name/src/datasources/TestDatasource.ts");

    expect(datasource).toMatchSnapshot();

    const dockerCompose = FakeCliFs.entries.get("project-name/docker-compose.yml");
    expect(dockerCompose).toMatchSnapshot();
  });
});
