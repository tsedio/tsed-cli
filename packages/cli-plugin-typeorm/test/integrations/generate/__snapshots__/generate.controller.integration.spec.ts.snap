// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate DataSource should generate the template with the right options (simple path) 1`] = `
"import {registerProvider} from \\"@tsed/di\\";
import {DataSource} from \\"typeorm\\";
import {Logger} from \\"@tsed/logger\\";

export const TEST_DATASOURCE = Symbol.for(\\"TestDatasource\\");
export const TestDatasource = new DataSource({
  type: \\"mysql\\",
  entities: [],
  host: \\"localhost\\",
  port: 3306,
  username: \\"test\\",
  password: \\"test\\",
  database: \\"test\\"
});

registerProvider<DataSource>({
  provide: TEST_DATASOURCE,
  type: \\"typeorm:datasource\\",
  deps: [Logger],
  async useAsyncFactory(logger: Logger) {
    await TestDatasource.initialize();

    logger.info(\\"Connected with typeorm to database: Test\\");

    return TestDatasource;
  },
  hooks: {
    $onDestroy(dataSource) {
      return dataSource.isInitialized && dataSource.close();
    }
  }
});
"
`;

exports[`Generate DataSource should generate the template with the right options (simple path) 2`] = `
"services:
  test:
    image: mysql:8.0.28-oracle
    ports:
      - '3306:3306'
    environment:
      MYSQL_ROOT_PASSWORD: admin
      MYSQL_USER: test
      MYSQL_PASSWORD: test
      MYSQL_DATABASE: test
"
`;
